src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs

```diff
@@ -374,6 +374,20 @@ public static void Sleep(int millisecondsTimeout)
         internal static ulong CurrentOSThreadId => GetCurrentOSThreadId();
 #endif
 
+        internal static bool ShouldSpinWait { get; } = DetermineShouldSpinWait();
+
+        // In cases where the cpu is limited due to cpu quota limits or there's only a single processor
+        // choosing not to spin wait might allow for more actual work to be done.
+        // However it's possible to spin wait in those scenarios by enabling the SpinWaitWhenCpuQuotaIsLimited config setting.
+        private static bool DetermineShouldSpinWait()
+        {
+            return !Environment.IsSingleProcessor &&
+                        (!Environment.IsCpuQuotaLimited ||
+                        AppContextConfigHelper.GetBooleanConfig(
+                        "System.Threading.ThreadPool.SpinWaitWhenCpuQuotaIsLimited",
+                        "DOTNET_ThreadPool_SpinWaitWhenCpuQuotaIsLimited"));
```

Comment:
Nit: indentation for clarity:
```suggestion
            return !Environment.IsSingleProcessor &&
                        (!Environment.IsCpuQuotaLimited ||
                        AppContextConfigHelper.GetBooleanConfig(
                            "System.Threading.ThreadPool.SpinWaitWhenCpuQuotaIsLimited",
                            "DOTNET_ThreadPool_SpinWaitWhenCpuQuotaIsLimited",
                            defaultValue: false));
```

It would also be more clear to explicitly pass in the default value, as some config vars can default to true, and the default value for the argument is not clear at all sites.

Personally, I think this is easier to read:
```suggestion
            return
                !Environment.IsSingleProcessor &&
                (
                    !Environment.IsCpuQuotaLimited ||
                    AppContextConfigHelper.GetBooleanConfig(
                        "System.Threading.ThreadPool.SpinWaitWhenCpuQuotaIsLimited",
                        "DOTNET_ThreadPool_SpinWaitWhenCpuQuotaIsLimited",
                        defaultValue: false)
                );
```